pkg/volume/quobyte/quobyte.go

```diff
@@ -0,0 +1,243 @@
+		plugin:  plugin,
+	}}, nil
+}
+
+func (plugin *quobytePlugin) execCommand(command string, args []string) ([]byte, error) {
+	cmd := plugin.exe.Command(command, args...)
+	return cmd.CombinedOutput()
+}
+
+// Quobyte volumes represent a bare host file or directory mount of an quobyte export.
+type quobyte struct {
+	volName string
+	pod     *api.Pod
+	mounter mount.Interface
+	plugin  *quobytePlugin
+	volume.MetricsNil
+}
+
+type quobyteMounter struct {
+	*quobyte
+	registry string
+	volume   string
+	path     string
+	readOnly bool
+	exe      exec.Interface
+}
+
+var _ volume.Mounter = &quobyteMounter{}
+
+func (mounter *quobyteMounter) GetAttributes() volume.Attributes {
+	return volume.Attributes{
+		ReadOnly:        mounter.readOnly,
+		Managed:         false,
+		SupportsSELinux: false,
+	}
+}
+
+// SetUp attaches the disk and bind mounts to the volume path.
+func (mounter *quobyteMounter) SetUp(fsGroup *int64) error {
+	return mounter.SetUpAt(mounter.GetPath(), fsGroup)
+}
+
+func (mounter *quobyteMounter) SetUpAt(dir string, fsGroup *int64) error {
+	notMnt, err := mounter.mounter.IsLikelyNotMountPoint(dir)
+	glog.V(4).Infof("quobyte: mount set up: %s %v %v", dir, !notMnt, err)
+	if err != nil && !os.IsNotExist(err) {
+		return err
+	}
+	if !notMnt {
+		return nil
+	}
+
+	os.MkdirAll(dir, 0750)
+	if err = mounter.setUpAtInternal(dir, fsGroup); err == nil {
+		return nil
+	}
+
+	// Cleanup upon failure.
+	c := &quobyteUnmounter{mounter.quobyte}
+	c.cleanup(dir)
+	return err
+}
+
+func (quobyteVolume *quobyte) GetPath() string {
+	return quobyteVolume.plugin.host.GetPodVolumeDir(quobyteVolume.pod.UID, strings.EscapeQualifiedNameForDisk(quobytePluginName), quobyteVolume.volName)
+}
+
+type quobyteUnmounter struct {
+	*quobyte
+}
+
+var _ volume.Unmounter = &quobyteUnmounter{}
+
+func (unmounter *quobyteUnmounter) TearDown() error {
+	return unmounter.TearDownAt(unmounter.GetPath())
+}
+
+func (unmounter *quobyteUnmounter) TearDownAt(dir string) error {
+	return unmounter.cleanup(dir)
+}
+
+func (unmounter *quobyteUnmounter) cleanup(dir string) error {
+	notMnt, err := unmounter.mounter.IsLikelyNotMountPoint(dir)
+	if err != nil {
+		return fmt.Errorf("quobyte: Error checking IsLikelyNotMountPoint: %v", err)
+	}
+	if notMnt {
+		return os.RemoveAll(dir)
+	}
+
+	if err := unmounter.mounter.Unmount(dir); err != nil {
+		return fmt.Errorf("quobyte: Unmounting failed: %v", err)
+	}
+	notMnt, mntErr := unmounter.mounter.IsLikelyNotMountPoint(dir)
+	if mntErr != nil {
+		return fmt.Errorf("quobyte: IsLikelyNotMountPoint check failed: %v", mntErr)
+	}
+	if notMnt {
+		if err := os.RemoveAll(dir); err != nil {
+			return fmt.Errorf("quobyte: RemoveAll failed: %v", err)
+		}
+	}
+
+	return nil
+}
+
+func (mounter *quobyteMounter) setUpAtInternal(dir string, fsGroup *int64) error {
+	var options []string
+	if mounter.readOnly {
+		options = append(options, "ro")
+	}
+
+	dirhost := path.Join(mounter.registry, mounter.volume, mounter.path)
```

Comment:
If the path is empty the whole volume get's mounted e.q.:
registry = registry:7861
volume = test
path = <empty>

This would result in `dirhost = registry:7861/test` so quobyte mounts the whole volume otherwise only the specified path inside the volume would be mounted.